### ISSUE TITLE
Middleware cause build to fail with next.js v13.4.4

### ISSUE THREAD

**Issue by alphaolomi** [nextauthjs/next-auth#7770]
### Environment

```yaml
  System:
    OS: macOS 13.4
    CPU: (4) x64 Intel(R) Core(TM) i5-7360U CPU @ 2.30GHz
    Memory: 295.57 MB / 8.00 GB
    Shell: 5.9 - /bin/zsh
  Binaries:
    Node: 20.2.0 - /usr/local/bin/node
    Yarn: 1.22.19 - /usr/local/bin/yarn
    npm: 9.6.7 - /usr/local/bin/npm
  Browsers:
    Brave Browser: 114.1.52.122
    Chrome: 113.0.5672.126
    Edge: 110.0.1587.63
    Safari: 16.5
```

### Reproduction URL

https://github.com/alphaolomi/nextjs-13-app-dir-next-auth-middleware-bug

### expected behavior

- build succeeds

### actual behavior

- build fails with error

```bash
❯ yrb | pbcopy                                                                                
Failed to compile.

./node_modules/.pnpm/next-auth@4.22.1_next@13.4.4_react-dom@18.2.0_react@18.2.0/node_modules/next-auth/next/middleware.js
Module parse failed: Identifier 'NextResponse' has already been declared (3:6)
File was processed with these loaders:
 * ./node_modules/.pnpm/next@13.4.4_react-dom@18.2.0_react@18.2.0/node_modules/next/dist/build/webpack/loaders/next-swc-loader.js
You may need an additional loader to handle the result of these loaders.
| "use strict";
| const NextResponse = require("next/dist/server/web/spec-extension/response").NextResponse;
> const NextResponse = require("next/dist/server/web/spec-extension/response").NextResponse;
| var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
| Object.defineProperty(exports, "__esModule", {

Import trace for requested module:
./node_modules/.pnpm/next-auth@4.22.1_next@13.4.4_react-dom@18.2.0_react@18.2.0/node_modules/next-auth/next/middleware.js
./node_modules/.pnpm/next-auth@4.22.1_next@13.4.4_react-dom@18.2.0_react@18.2.0/node_modules/next-auth/middleware.js


> Build failed because of webpack errors
error Command failed with exit code 1.   
```

### How to reproduce

- Create new Next.js app using CNA `create-next-app --version` => 13.4.4
- Add `next-auth` `"next-auth": "4.22.1",` in package.json
- Configure [Credentials provider](https://next-auth.js.org/configuration/providers/credentials)  in app dir
- Configure Middleware see [Next.js <> NextAuth middleware docs](https://next-auth.js.org/configuration/nextjs#middleware)
- Run `yarn run build` or `pnpm run build`

### Expected behavior

- build succeeds

**Comment by davecarlson** [comment]
This was a next problem i believe, fixed in 13.4.5

**Comment by balazsorban44** [comment]
Closing as per the comment above.
